### ISSUE TITLE
CreateTun error if TunConfig.Ipv6 != ""

### ISSUE THREAD

**Issue by ssucker** [kubenetworks/kubevpn#119]
I run kubevpn proxy command `kubevpn proxy deploymeny/xx`, 
```yaml
PS D:\kubevpn\bin> kubevpn proxy deployment/test-deploy --headers name=jack
connectting to cluster
start to connect
got cidr from cache
get cidr successfully
update ref count successfully
traffic manager already exist, reuse it
port forward ready
add ipv6 addr to tun err:Element not found.
create tun device error: Element not found.
create tun listener error: Element not found.
parse route error: Element not found.
start local tun service failed: Element not found.
do connect error: Element not found.
prepare to exit, cleaning up
update ref count successfully
clean up successfully
Error: rpc error: code = Unknown desc = Element not found.
```
error occurs in tun_windows.go:createTun func:  `ifName.AddIPAddress(prefix)`
```golang
func createTun(cfg Config) (conn net.Conn, itf *net.Interface, err error) {
	if cfg.Addr == "" && cfg.Addr6 == "" {
		err = fmt.Errorf("ipv4 address and ipv6 address can not be empty at same time")
		return
	}

	interfaceName := "kubevpn"
	if len(cfg.Name) != 0 {
		interfaceName = cfg.Name
	}
	tunDevice, err := wireguardtun.CreateTUN(interfaceName, cfg.MTU)
	if err != nil {
		err = fmt.Errorf("failed to create TUN device: %w", err)
		return
	}

	ifName := winipcfg.LUID(tunDevice.(*wireguardtun.NativeTun).LUID())

	var ipv4, ipv6 net.IP
	if cfg.Addr != "" {
		if ipv4, _, err = net.ParseCIDR(cfg.Addr); err != nil {
			return
		}
		var prefix netip.Prefix
		if prefix, err = netip.ParsePrefix(cfg.Addr); err != nil {
			return
		}
		if err = ifName.AddIPAddress(prefix); err != nil {
			return
		}
	}

	if cfg.Addr6 != "" {
		if ipv6, _, err = net.ParseCIDR(cfg.Addr6); err != nil {
			return
		}
		var prefix netip.Prefix
		if prefix, err = netip.ParsePrefix(cfg.Addr6); err != nil {
			return
		}
		if err = ifName.AddIPAddress(prefix); err != nil {
			log.Infof("add ipv6 addr to tun err:%v", err)
			return
		}
	}

```
i used nocalhost's proxy mode before, and nocalhost does not support ipv6, but it works for me.
if i remove the setting env code:
```golang
func (c *ConnectOptions) startLocalTunServe(ctx context.Context, forwardAddress string, lite bool) (err error) {
	// todo figure it out why
	if util.IsWindows() {
		c.localTunIPv4.Mask = net.CIDRMask(0, 32)
	}
	var list = sets.New[string]()
	if !lite {
		list.Insert(config.CIDR.String())
	}
	for _, ipNet := range c.cidrs {
		list.Insert(ipNet.String())
	}
	// add extra-cidr
	for _, s := range c.ExtraCIDR {
		_, _, err = net.ParseCIDR(s)
		if err != nil {
			return fmt.Errorf("invalid extra-cidr %s, err: %v", s, err)
		}
		list.Insert(s)
	}
	// if err = os.Setenv(config.EnvInboundPodTunIPv6, c.localTunIPv6.String()); err != nil {
	//	return err
	// }
```
createTun will work successfully.
does anyone know how to solve this problem?

**Comment by wencaiwulue** [comment]
Hello, sorry for you meet this issue. Can you privode which Windows version you are using?

**Comment by wencaiwulue** [comment]
`D:\>kubevpn proxy deployment/authors --headers a=1
connectting to cluster
start to connect
got cidr from cache
get cidr successfully
update ref count successfully
traffic manager already exist, reuse it
port forward ready
tunnel connected
adding route...
dns service ok
start to create remote inbound pod for deployment/authors
patch workload default/deployment/authors with sidecar
rollout status for deployment/authors
Waiting for deployment spec update to be observed...
Waiting for deployment spec update to be observed...
Waiting for deployment "authors" rollout to finish: 0 out of 1 new replicas have been updated...
Waiting for deployment "authors" rollout to finish: 1 old replicas are pending termination...
Waiting for deployment "authors" rollout to finish: 1 old replicas are pending termination...
deployment "authors" successfully rolled out
rollout status for deployment/authors successfully
create remote inbound pod for deployment/authors successfully
+---------------------------------------------------------------------------+
|    Now you can access resources in the kubernetes cluster, enjoy it :)    |
+---------------------------------------------------------------------------+

D:\>ipconfig

Windows IP 配置


未知适配器 kubevpn:

   连接特定的 DNS 后缀 . . . . . . . : default.svc.cluster.local
   IPv6 地址 . . . . . . . . . . . . : efff:ffff:ffff:ffff:ffff:ffff:ffff:99a0
   本地链接 IPv6 地址. . . . . . . . : fe80::f4fc:fdda:62c3:7940%11
   IPv4 地址 . . . . . . . . . . . . : 223.254.0.108
   子网掩码  . . . . . . . . . . . . : 0.0.0.0
   默认网关. . . . . . . . . . . . . :
` I tested it on Windows10, it looks very well

**Comment by ssucker** [comment]
> Hello, sorry for you meet this issue. Can you privode which Windows version you are using? and i searched `Element not found` issue on Google, It shows that please make sure `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Tcpip6\Parameters\DisabledComponents ` this key should not exist. [ref](https://forums.he.net/index.php?topic=653.0) [ref2](https://forums.he.net/index.php?topic=1492.0)

My os is:
版本	Windows 10 专业版
版本号	21H1
安装日期	‎2023/‎5/‎30
操作系统内部版本	19043.1645
体验	Windows Feature Experience Pack 120.2212.4170.0

**Comment by wencaiwulue** [comment]
> > Hello, sorry for you meet this issue. Can you privode which Windows version you are using? and i searched `Element not found` issue on Google, It shows that please make sure `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Tcpip6\Parameters\DisabledComponents ` this key should not exist. [ref](https://forums.he.net/index.php?topic=653.0) [ref2](https://forums.he.net/index.php?topic=1492.0)
> 
> My os is: 版本 Windows 10 专业版 版本号 21H1 安装日期 ‎2023/‎5/‎30 操作系统内部版本 19043.1645 体验 Windows Feature Experience Pack 120.2212.4170.0

Ok, got it, can you please check the bellow link to make sure registry table is setting correctly
[https://forums.he.net/index.php?topic=653.0](https://forums.he.net/index.php?topic=653.0) [https://forums.he.net/index.php?topic=1492.0](https://forums.he.net/index.php?topic=1492.0)

**Comment by ssucker** [comment]
> > > Hello, sorry for you meet this issue. Can you privode which Windows version you are using? and i searched `Element not found` issue on Google, It shows that please make sure `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Tcpip6\Parameters\DisabledComponents ` this key should not exist. [ref](https://forums.he.net/index.php?topic=653.0) [ref2](https://forums.he.net/index.php?topic=1492.0)
> > 
> > 
> > My os is: 版本 Windows 10 专业版 版本号 21H1 安装日期 ‎2023/‎5/‎30 操作系统内部版本 19043.1645 体验 Windows Feature Experience Pack 120.2212.4170.0
> 
> Ok, got it, can you please check the bellow link to make sure registry table is setting correctly https://forums.he.net/index.php?topic=653.0 https://forums.he.net/index.php?topic=1492.0

OK, i'll give it a try.

**Comment by ssucker** [comment]
> > > Hello, sorry for you meet this issue. Can you privode which Windows version you are using? and i searched `Element not found` issue on Google, It shows that please make sure `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Tcpip6\Parameters\DisabledComponents ` this key should not exist. [ref](https://forums.he.net/index.php?topic=653.0) [ref2](https://forums.he.net/index.php?topic=1492.0)
> > 
> > 
> > My os is: 版本 Windows 10 专业版 版本号 21H1 安装日期 ‎2023/‎5/‎30 操作系统内部版本 19043.1645 体验 Windows Feature Experience Pack 120.2212.4170.0
> 
> Ok, got it, can you please check the bellow link to make sure registry table is setting correctly https://forums.he.net/index.php?topic=653.0 https://forums.he.net/index.php?topic=1492.0

It works! Thanks a lot!

**Comment by wencaiwulue** [comment]
> > > > Hello, sorry for you meet this issue. Can you privode which Windows version you are using? and i searched `Element not found` issue on Google, It shows that please make sure `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Tcpip6\Parameters\DisabledComponents ` this key should not exist. [ref](https://forums.he.net/index.php?topic=653.0) [ref2](https://forums.he.net/index.php?topic=1492.0)
> > > 
> > > 
> > > My os is: 版本 Windows 10 专业版 版本号 21H1 安装日期 ‎2023/‎5/‎30 操作系统内部版本 19043.1645 体验 Windows Feature Experience Pack 120.2212.4170.0
> > 
> > 
> > Ok, got it, can you please check the bellow link to make sure registry table is setting correctly https://forums.he.net/index.php?topic=653.0 https://forums.he.net/index.php?topic=1492.0
> 
> It works! Thanks a lot!

Wow, congratulations ~~. Maybe you can introduce kubevpn to your colleagues or friends, welcome pr and star ~

**Comment by ssucker** [comment]
Yes, i will.

**Comment by kyr1eeee** [comment]
> Hello, sorry for you meet this issue. Can you privode which Windows version you are using? and i searched `Element not found` issue on Google, It shows that please make sure `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Tcpip6\Parameters\DisabledComponents ` this key should not exist. [ref](https://forums.he.net/index.php?topic=653.0) [ref2](https://forums.he.net/index.php?topic=1492.0)

sorry to bother, but i wonder what is the meaning of adding ipv6 address to tun device in `pkg/tun/tun_windows.go:CreateTun()`

```golang
	if cfg.Addr6 != "" {
		if ipv6, _, err = net.ParseCIDR(cfg.Addr6); err != nil {
			return
		}
		var prefix netip.Prefix
		if prefix, err = netip.ParsePrefix(cfg.Addr6); err != nil {
			return
		}
		if err = ifName.AddIPAddress(prefix); err != nil {
			return
		}
	}
```
it seems works fine when tun device has no ipv6 address

**Comment by wencaiwulue** [comment]
> > Hello, sorry for you meet this issue. Can you privode which Windows version you are using? and i searched `Element not found` issue on Google, It shows that please make sure `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Tcpip6\Parameters\DisabledComponents ` this key should not exist. [ref](https://forums.he.net/index.php?topic=653.0) [ref2](https://forums.he.net/index.php?topic=1492.0)
> 
> sorry to bother, but i wonder what is the meaning of adding ipv6 address to tun device in `pkg/tun/tun_windows.go:CreateTun()`
> 
> ```go
> 	if cfg.Addr6 != "" {
> 		if ipv6, _, err = net.ParseCIDR(cfg.Addr6); err != nil {
> 			return
> 		}
> 		var prefix netip.Prefix
> 		if prefix, err = netip.ParsePrefix(cfg.Addr6); err != nil {
> 			return
> 		}
> 		if err = ifName.AddIPAddress(prefix); err != nil {
> 			return
> 		}
> 	}
> ```
> 
> it seems works fine when tun device has no ipv6 address

It will generate a IPv6 addr by default. so our computer needs to enable IPv6 feature

**Comment by kyr1eeee** [comment]
> > > Hello, sorry for you meet this issue. Can you privode which Windows version you are using? and i searched `Element not found` issue on Google, It shows that please make sure `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Tcpip6\Parameters\DisabledComponents ` this key should not exist. [ref](https://forums.he.net/index.php?topic=653.0) [ref2](https://forums.he.net/index.php?topic=1492.0)
> > 
> > 
> > sorry to bother, but i wonder what is the meaning of adding ipv6 address to tun device in `pkg/tun/tun_windows.go:CreateTun()`
> > ```go
> > 	if cfg.Addr6 != "" {
> > 		if ipv6, _, err = net.ParseCIDR(cfg.Addr6); err != nil {
> > 			return
> > 		}
> > 		var prefix netip.Prefix
> > 		if prefix, err = netip.ParsePrefix(cfg.Addr6); err != nil {
> > 			return
> > 		}
> > 		if err = ifName.AddIPAddress(prefix); err != nil {
> > 			return
> > 		}
> > 	}
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > it seems works fine when tun device has no ipv6 address
> 
> It will generate a IPv6 addr by default. so our computer needs to enable IPv6 feature

ok, got it.

**Comment by wencaiwulue** [comment]
> > > > Hello, sorry for you meet this issue. Can you privode which Windows version you are using? and i searched `Element not found` issue on Google, It shows that please make sure `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Tcpip6\Parameters\DisabledComponents ` this key should not exist. [ref](https://forums.he.net/index.php?topic=653.0) [ref2](https://forums.he.net/index.php?topic=1492.0)
> > > 
> > > 
> > > sorry to bother, but i wonder what is the meaning of adding ipv6 address to tun device in `pkg/tun/tun_windows.go:CreateTun()`
> > > ```go
> > > 	if cfg.Addr6 != "" {
> > > 		if ipv6, _, err = net.ParseCIDR(cfg.Addr6); err != nil {
> > > 			return
> > > 		}
> > > 		var prefix netip.Prefix
> > > 		if prefix, err = netip.ParsePrefix(cfg.Addr6); err != nil {
> > > 			return
> > > 		}
> > > 		if err = ifName.AddIPAddress(prefix); err != nil {
> > > 			return
> > > 		}
> > > 	}
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > it seems works fine when tun device has no ipv6 address
> > 
> > 
> > It will generate a IPv6 addr by default. so our computer needs to enable IPv6 feature
> 
> ok, got it.

Microsoft docs:
https://learn.microsoft.com/en-us/troubleshoot/windows-server/networking/configure-ipv6-in-windows#use-registry-key-to-configure-ipv6
https://learn.microsoft.com/zh-cn/troubleshoot/windows-server/networking/configure-ipv6-in-windows#use-registry-key-to-configure-ipv6

**Comment by kyr1eeee** [comment]
> > > > > Hello, sorry for you meet this issue. Can you privode which Windows version you are using? and i searched `Element not found` issue on Google, It shows that please make sure `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Tcpip6\Parameters\DisabledComponents ` this key should not exist. [ref](https://forums.he.net/index.php?topic=653.0) [ref2](https://forums.he.net/index.php?topic=1492.0)
> > > > 
> > > > 
> > > > sorry to bother, but i wonder what is the meaning of adding ipv6 address to tun device in `pkg/tun/tun_windows.go:CreateTun()`
> > > > ```go
> > > > 	if cfg.Addr6 != "" {
> > > > 		if ipv6, _, err = net.ParseCIDR(cfg.Addr6); err != nil {
> > > > 			return
> > > > 		}
> > > > 		var prefix netip.Prefix
> > > > 		if prefix, err = netip.ParsePrefix(cfg.Addr6); err != nil {
> > > > 			return
> > > > 		}
> > > > 		if err = ifName.AddIPAddress(prefix); err != nil {
> > > > 			return
> > > > 		}
> > > > 	}
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > it seems works fine when tun device has no ipv6 address
> > > 
> > > 
> > > It will generate a IPv6 addr by default. so our computer needs to enable IPv6 feature
> > 
> > 
> > ok, got it.
> 
> Microsoft docs: https://learn.microsoft.com/en-us/troubleshoot/windows-server/networking/configure-ipv6-in-windows#use-registry-key-to-configure-ipv6 https://learn.microsoft.com/zh-cn/troubleshoot/windows-server/networking/configure-ipv6-in-windows#use-registry-key-to-configure-ipv6

Is that possible that i do not need ipv6 capability, and tun device has no need to add an ipv6 address.
I have this issue because somehow me and my friend both encoutered the "element not found" error, we two must configure  the ipv6 registry key. But actually we seem don't want ipv6 feature so much. So could the ipv6 capability be an user option when use kubevpn proxy?

**Comment by wencaiwulue** [comment]
> > > > > > Hello, sorry for you meet this issue. Can you privode which Windows version you are using? and i searched `Element not found` issue on Google, It shows that please make sure `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Tcpip6\Parameters\DisabledComponents ` this key should not exist. [ref](https://forums.he.net/index.php?topic=653.0) [ref2](https://forums.he.net/index.php?topic=1492.0)
> > > > > 
> > > > > 
> > > > > sorry to bother, but i wonder what is the meaning of adding ipv6 address to tun device in `pkg/tun/tun_windows.go:CreateTun()`
> > > > > ```go
> > > > > 	if cfg.Addr6 != "" {
> > > > > 		if ipv6, _, err = net.ParseCIDR(cfg.Addr6); err != nil {
> > > > > 			return
> > > > > 		}
> > > > > 		var prefix netip.Prefix
> > > > > 		if prefix, err = netip.ParsePrefix(cfg.Addr6); err != nil {
> > > > > 			return
> > > > > 		}
> > > > > 		if err = ifName.AddIPAddress(prefix); err != nil {
> > > > > 			return
> > > > > 		}
> > > > > 	}
> > > > > ```
> > > > > 
> > > > > 
> > > > >     
> > > > >       
> > > > >     
> > > > > 
> > > > >       
> > > > >     
> > > > > 
> > > > >     
> > > > >   
> > > > > it seems works fine when tun device has no ipv6 address
> > > > 
> > > > 
> > > > It will generate a IPv6 addr by default. so our computer needs to enable IPv6 feature
> > > 
> > > 
> > > ok, got it.
> > 
> > 
> > Microsoft docs: https://learn.microsoft.com/en-us/troubleshoot/windows-server/networking/configure-ipv6-in-windows#use-registry-key-to-configure-ipv6 https://learn.microsoft.com/zh-cn/troubleshoot/windows-server/networking/configure-ipv6-in-windows#use-registry-key-to-configure-ipv6
> 
> Is that possible that i do not need ipv6 capability, and tun device has no need to add an ipv6 address. I have this issue because somehow me and my friend both encoutered the "element not found" error, we two must configure the ipv6 registry key. But actually we seem don't want ipv6 feature so much. So could the ipv6 capability be an user option when use kubevpn proxy?

Maybe i can check if PC not enable IPv6, not to set IPv6 address ? 🤔

**Comment by kyr1eeee** [comment]
> > > > > > > Hello, sorry for you meet this issue. Can you privode which Windows version you are using? and i searched `Element not found` issue on Google, It shows that please make sure `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Tcpip6\Parameters\DisabledComponents ` this key should not exist. [ref](https://forums.he.net/index.php?topic=653.0) [ref2](https://forums.he.net/index.php?topic=1492.0)
> > > > > > 
> > > > > > 
> > > > > > sorry to bother, but i wonder what is the meaning of adding ipv6 address to tun device in `pkg/tun/tun_windows.go:CreateTun()`
> > > > > > ```go
> > > > > > 	if cfg.Addr6 != "" {
> > > > > > 		if ipv6, _, err = net.ParseCIDR(cfg.Addr6); err != nil {
> > > > > > 			return
> > > > > > 		}
> > > > > > 		var prefix netip.Prefix
> > > > > > 		if prefix, err = netip.ParsePrefix(cfg.Addr6); err != nil {
> > > > > > 			return
> > > > > > 		}
> > > > > > 		if err = ifName.AddIPAddress(prefix); err != nil {
> > > > > > 			return
> > > > > > 		}
> > > > > > 	}
> > > > > > ```
> > > > > > 
> > > > > > 
> > > > > >     
> > > > > >       
> > > > > >     
> > > > > > 
> > > > > >       
> > > > > >     
> > > > > > 
> > > > > >     
> > > > > >   
> > > > > > it seems works fine when tun device has no ipv6 address
> > > > > 
> > > > > 
> > > > > It will generate a IPv6 addr by default. so our computer needs to enable IPv6 feature
> > > > 
> > > > 
> > > > ok, got it.
> > > 
> > > 
> > > Microsoft docs: https://learn.microsoft.com/en-us/troubleshoot/windows-server/networking/configure-ipv6-in-windows#use-registry-key-to-configure-ipv6 https://learn.microsoft.com/zh-cn/troubleshoot/windows-server/networking/configure-ipv6-in-windows#use-registry-key-to-configure-ipv6
> > 
> > 
> > Is that possible that i do not need ipv6 capability, and tun device has no need to add an ipv6 address. I have this issue because somehow me and my friend both encoutered the "element not found" error, we two must configure the ipv6 registry key. But actually we seem don't want ipv6 feature so much. So could the ipv6 capability be an user option when use kubevpn proxy?
> 
> Maybe i can check if PC not enable IPv6, not to set IPv6 address ? 🤔

i have tried this case before, it seems to work well.

**Comment by wencaiwulue** [comment]
> > > > > > > > Hello, sorry for you meet this issue. Can you privode which Windows version you are using? and i searched `Element not found` issue on Google, It shows that please make sure `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Tcpip6\Parameters\DisabledComponents ` this key should not exist. [ref](https://forums.he.net/index.php?topic=653.0) [ref2](https://forums.he.net/index.php?topic=1492.0)
> > > > > > > 
> > > > > > > 
> > > > > > > sorry to bother, but i wonder what is the meaning of adding ipv6 address to tun device in `pkg/tun/tun_windows.go:CreateTun()`
> > > > > > > ```go
> > > > > > > 	if cfg.Addr6 != "" {
> > > > > > > 		if ipv6, _, err = net.ParseCIDR(cfg.Addr6); err != nil {
> > > > > > > 			return
> > > > > > > 		}
> > > > > > > 		var prefix netip.Prefix
> > > > > > > 		if prefix, err = netip.ParsePrefix(cfg.Addr6); err != nil {
> > > > > > > 			return
> > > > > > > 		}
> > > > > > > 		if err = ifName.AddIPAddress(prefix); err != nil {
> > > > > > > 			return
> > > > > > > 		}
> > > > > > > 	}
> > > > > > > ```
> > > > > > > 
> > > > > > > 
> > > > > > >     
> > > > > > >       
> > > > > > >     
> > > > > > > 
> > > > > > >       
> > > > > > >     
> > > > > > > 
> > > > > > >     
> > > > > > >   
> > > > > > > it seems works fine when tun device has no ipv6 address
> > > > > > 
> > > > > > 
> > > > > > It will generate a IPv6 addr by default. so our computer needs to enable IPv6 feature
> > > > > 
> > > > > 
> > > > > ok, got it.
> > > > 
> > > > 
> > > > Microsoft docs: https://learn.microsoft.com/en-us/troubleshoot/windows-server/networking/configure-ipv6-in-windows#use-registry-key-to-configure-ipv6 https://learn.microsoft.com/zh-cn/troubleshoot/windows-server/networking/configure-ipv6-in-windows#use-registry-key-to-configure-ipv6
> > > 
> > > 
> > > Is that possible that i do not need ipv6 capability, and tun device has no need to add an ipv6 address. I have this issue because somehow me and my friend both encoutered the "element not found" error, we two must configure the ipv6 registry key. But actually we seem don't want ipv6 feature so much. So could the ipv6 capability be an user option when use kubevpn proxy?
> > 
> > 
> > Maybe i can check if PC not enable IPv6, not to set IPv6 address ? 🤔
> 
> i have tried this case before, it seems to work well.

Ok. I got it. I will fix it as soon as possible.

**Comment by wencaiwulue** [comment]
> > > > > > > > > Hello, sorry for you meet this issue. Can you privode which Windows version you are using? and i searched `Element not found` issue on Google, It shows that please make sure `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Tcpip6\Parameters\DisabledComponents ` this key should not exist. [ref](https://forums.he.net/index.php?topic=653.0) [ref2](https://forums.he.net/index.php?topic=1492.0)
> > > > > > > > 
> > > > > > > > 
> > > > > > > > sorry to bother, but i wonder what is the meaning of adding ipv6 address to tun device in `pkg/tun/tun_windows.go:CreateTun()`
> > > > > > > > ```go
> > > > > > > > 	if cfg.Addr6 != "" {
> > > > > > > > 		if ipv6, _, err = net.ParseCIDR(cfg.Addr6); err != nil {
> > > > > > > > 			return
> > > > > > > > 		}
> > > > > > > > 		var prefix netip.Prefix
> > > > > > > > 		if prefix, err = netip.ParsePrefix(cfg.Addr6); err != nil {
> > > > > > > > 			return
> > > > > > > > 		}
> > > > > > > > 		if err = ifName.AddIPAddress(prefix); err != nil {
> > > > > > > > 			return
> > > > > > > > 		}
> > > > > > > > 	}
> > > > > > > > ```
> > > > > > > > 
> > > > > > > > 
> > > > > > > >     
> > > > > > > >       
> > > > > > > >     
> > > > > > > > 
> > > > > > > >       
> > > > > > > >     
> > > > > > > > 
> > > > > > > >     
> > > > > > > >   
> > > > > > > > it seems works fine when tun device has no ipv6 address
> > > > > > > 
> > > > > > > 
> > > > > > > It will generate a IPv6 addr by default. so our computer needs to enable IPv6 feature
> > > > > > 
> > > > > > 
> > > > > > ok, got it.
> > > > > 
> > > > > 
> > > > > Microsoft docs: https://learn.microsoft.com/en-us/troubleshoot/windows-server/networking/configure-ipv6-in-windows#use-registry-key-to-configure-ipv6 https://learn.microsoft.com/zh-cn/troubleshoot/windows-server/networking/configure-ipv6-in-windows#use-registry-key-to-configure-ipv6
> > > > 
> > > > 
> > > > Is that possible that i do not need ipv6 capability, and tun device has no need to add an ipv6 address. I have this issue because somehow me and my friend both encoutered the "element not found" error, we two must configure the ipv6 registry key. But actually we seem don't want ipv6 feature so much. So could the ipv6 capability be an user option when use kubevpn proxy?
> > > 
> > > 
> > > Maybe i can check if PC not enable IPv6, not to set IPv6 address ? 🤔
> > 
> > 
> > i have tried this case before, it seems to work well.
> 
> Ok. I got it. I will fix it as soon as possible.

fixed this issue in commit: f7957c2a25f52a0af68c3af9154c3514a2a538ad

**Comment by wencaiwulue** [comment]
@kyr1eeee. Hello. i redo released v2.2.1 and fixed this issue. You can download latest version and try it again. Or us e command `kubevpn upgrade`   to upgrade

**Comment by kyr1eeee** [comment]
Yes, it works even if my pc doesn't enable tcpip6
